### PR TITLE
Fix behaviour of hasUppercase and hasLowercase

### PR DESCRIPTION
**Pull request by 2mt-heuser**
Regarding https://github.com/zestframework/Zest_Framework/issues/274 how about this implementation of hasUpper/Lower?

Would that have any drawbacks?

### CODE REVIEW

**Review by lablnet — APPROVED** [review]

File: src/Data/Str.php [hunk 1]
```diff
@@ -103,43 +103,27 @@ public static function count(string $str, $encoding = null)
...
      * Check if string has atleast one uppercase.
      *
-     * @param string $str String to be checked.
+     * @param string      $str      String to be checked.
+     * @param string|null $encoding Valid encoding.
      *
      * @return bool
      */
-    public static function hasUpperCase(string $str)
+    public static function hasUpperCase(string $str, string $encoding = null): bool
```
Comment (peter279k) [inline src/Data/Str.php:111]: How about using following code?

```php
public static function hasUpperCase(string $str, ?string $encoding = null): bool
```

Comment (2mt-heuser) [reply src/Data/Str.php:111]: would have the exact same call-signature / implication?

when providing the default ```$encoding = null``` we don't need the ? to mark is as string|null

but I could change it to be in line with count() etc where there's no typehint string and only the default value if you'd prefer that.
```php
public static function reverse(string $str, $encoding = null) :string
public static function count(string $str, $encoding = null)
```

or add typehints to all the functions so they're inline with each other

Comment (peter279k) [reply src/Data/Str.php:111]: Hi, thanks for your explanation.

I don't prefer any of this type hints and @Lablnet may decide to use.

I care about that is about consistency.

Once we decide to use one of type hint style. It should be used on whole repository.

Comment (lablnet) [reply src/Data/Str.php:111]: I think we can continue as we are going with default type hinting..

Comment (2mt-heuser) [reply src/Data/Str.php:111]: With 7.4 and typed properties around the corner I kinda like transitioning to everything typed but nontheless, your choice. If you want them gone, it's just one commit from me away :-)

Comment (lablnet) [reply src/Data/Str.php:111]: No thanks for your time am going to Merger this PR, but lets first approve by @peter279k

**Review by peter279k — APPROVED** [review]

**Comment by lablnet** [comment]
Thanks you so much for your time .

**Pull request merged**